### ISSUE TITLE
Documentation on hover for intrinsic functions not working

### ISSUE THREAD

**Issue by maroony** [krvajal/vscode-fortran-support#250]
I do *not* use other fortran plugins and this feature doens't work for release 2.2.2 (= 2.3.0 Marketplace?).

Example (documentation for intrinsic function `max` doesn't show up):

```f90
module my

  implicit none

contains

  subroutine routine(arg)
    implicit none
    integer :: arg

    arg = max(arg, 0)

  end subroutine routine

end module my


```

**Comment by jefflarkin** [comment]
I'm seeing the same thing. In my local copy, if I copy `src/docs` to `out/src/docs` and rebuild, I start getting the pop-ups, so it may be a simple build issue. I will note though that the first time I hover over `max` I get an error in the debug console and I don't know the code well enough to know whether it's something I would have caused by making this copy. I do get the pop-up despite the error. The error comes from bootstrap-fork.js:5.

**Comment by gnikit** [comment]
@maroony In general the vscode-fortran-support extension for a complete IDE-like experience is meant to be used with the [fortran-language-server](https://github.com/hansec/fortran-language-server) and the accompanying VSCode extension [FORTRAN IntelliSense](https://marketplace.visualstudio.com/items?itemName=hansec.fortran-ls). The language server which is parsed into vscode through the FORTRAN IntelliSense extension provides substantially more features, liker complete hover support, go-tos and peek definitions and many more. For these reasons it is rarely recommended to use the inbuilt hover support since it is limited to Fortran intrinsics we have copied of the gfortran documentation. You can enable/disable the inbuilt hover support with `"fortran.provideHover": true,`

Now @jefflarkin I think you are onto something. Even though we rarely use the inbuilt hover it ought to still be functional. The offending line of code is https://github.com/krvajal/vscode-fortran-support/blob/0b6021126d4b2f72d258b7700b962d5f8a0afc95/src/lib/helper.ts#L42 where it looks for the `docs` inside `out/src` and it fails to find it so in the end does nothing.

There also appears to be a problem with parsing the .json documentation files into html.

**Comment by jefflarkin** [comment]
@gnikit Yes, this is where I tripped across my workaround. Seems like if the build process were slightly modified to copy those files into the out directory, then that file would cause the built-in hover to work.

I wasn't aware of the suggestion to install the other extensions either. Maybe the documentation could be improved to make that more obvious?

**Comment by maroony** [comment]
@gnikit Thanks for your feedback. So, maybe the whole feature could be removed in this extension? I'll give Fortran IntelliSense a try.

**Comment by gnikit** [comment]
Yes, I will have a look into it. I have been a bit busy lately with PhD submission closing in and I will admit, it has been a while since I wrote any TypeScript. 
I don't think we can simply call `cp` when building since it is not platform independent. I seem to recall a package called `ncp` that might be of use. 


> I wasn't aware of the suggestion

I realised that too. I will add a notify button or something + update README. There is an open PR where we are adding small bits and pieces such as this.

**Comment by gnikit** [comment]
So the feature still works (without the bug jeff mentioned) but with very limited support, compared to what a full language server can provide. I would be reluctant to remove because then we would have no hover functionality, and  since we are not the only Fortran extension out there we can't simply "copy" the Fortran IntelliSense code.

**Comment by gnikit** [comment]
A bit more on my initial comment about needing Fortran IntelliSense for integration with the Language Server.

That is not entirely true. A while ago we were able to interface with `fortls` directly, the user would have to pass the options to `fortls` either through `languageServerFlags` or with a `.fortls` file. 

From what I recall, around 2 years ago (see #121) the 2 extensions became incompatible with one another so the decision was made to disable `vscode-fortran-support`'s interface with `fortls` and leave it completely up to `Fortran IntelliSense` who was created by the author of `fortls`.

Our interface still works, you can test it yourself by setting the if conditional to `true` in https://github.com/krvajal/vscode-fortran-support/blob/0b6021126d4b2f72d258b7700b962d5f8a0afc95/src/extension.ts#L56 and supplying a `.fortls` in the root of your Extension Development Host like this one

```json
{
  "variable_hover": true,
  "hover_signature": true,
  "lowercase_intrinsics": true,
  "use_signature_help": true,
  "enable_code_actions": true
}
```

Given that the Fortran IntelliSense Extension has not been updated for quite some time, it might be worth reintroducing our own `fortls` interface.

**Comment by gnikit** [comment]
This is now fixed in PR #244 and will be included in the next release.
